Model/Quote/QuoteAction/Currency.php

```diff
@@ -39,19 +40,20 @@ public function getActionData(CartInterface $cart): array
 
         return [
             [
-                'type' => self::SET_CURRENCY,
+                'type' => self::SET_DISPLAY_CURRENCY,
```

Comment:
Discussion: Instead of calling this display currency should we work with base currency and charged currency (which would be the display currency). The base currency will be the shop currency and charged currency would be the cart currency. This way we will use the same wording everywhere across our services.